Changelog.md

```diff
@@ -9,6 +9,8 @@
     A trailing `finally` clause to `try`/`catch` expressions facilitates structured
     resource deallocation (e.g. acquired locks, etc.) and similar cleanups in the
     presence of control-flow expressions (`return`, `break`, `continue`, `throw`).
+    Additionally, in presence of `finally` the `catch` clause becomes optional and
+    potential errors fall through to any enclosing `catch`.
```

Comment:
```suggestion
    Additionally, in presence of `finally` the `catch` clause becomes optional and
    and any uncaught error from the `try` block will be propagated, after executing the `finally` block.
```

Its probably worth adding a test for this if we don't alread have one. i.e. Somethink like:
```
var done = false
try { throw Prim.error(foo) }
finally { done := true }
assert done
```